draft-ietf-tls-tls13.md

```diff
@@ -2791,12 +2805,34 @@ The following rules apply to the certificates sent by the server:
   guide certificate selection. As servers MAY require the presence of the server_name
   extension, clients SHOULD send this extension.
 
-All certificates provided by the server MUST be signed by a hash/signature
-algorithm pair that appears in the "signature_algorithms" extension provided
-by the client (see {{signature-algorithms}}).
-[[OPEN ISSUE: changing this is under consideration]]
-Note that this implies that a certificate containing a key for one signature
-algorithm MAY be signed using a different signature algorithm (for instance,
+All certificates provided by the server MUST be signed by a
+hash/signature algorithm pair that appears in the "signature_algorithms"
+extension provided by the client, if they are able to provide such
+a chain (see {{signature-algorithms}}).
+If the server cannot produce a certificate chain that is signed only via the
+indicated supported pairs, then it SHOULD continue the handshake by sending
+the client a certificate chain of its choice that may include algorithms
+that are not known to be supported by the client. This fallback chain MAY
+use the deprecated SHA hash algorithms, SHA-1 or SHA-224.
+If the client cannot construct an acceptable chain using the provided
+certificates and decides to abort the handshake, then it MUST send an
+"unsupported_certificate" alert message and close the connection.
+
+Any endpoint receiving any certificate signed using any signature algorithm
+using an MD5 hash MUST send a "bad_certificate" alert message and close
+the connection.
+
+Endpoints that offer support for SHA-1 or SHA-224 via the
+"signature_algorithms" extension SHOULD accept a chain that makes use of it.
+As these algorithms are deprecated, support for them is NOT RECOMMENDED.
```

Comment:
This seems simultaneously too strong and too weak. If you're going to reject
a certificate just because it has SHA-1 or SHA-224 then you shouldn't advertise
them at all (so SHOULD is too weak). However, there's no obligation not to reject the cert for other reasons (so SHOULD is too strong).